Source/Public/Set-VSTeamAgentPoolMaintenance.ps1

```diff
@@ -0,0 +1,97 @@
+function Set-VSTeamAgentPoolMaintenance {
+   [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = "Medium", DefaultParameterSetName = "Disabled",
+      HelpUri = 'https://methodsandpractices.github.io/vsteam-docs/docs/modules/vsteam/commands/Set-VSTeamAgentPoolMaintenance')]
+   param(
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true, Position = 1)]
+      [Alias('PoolID')]
+      [int] $Id,
+
+      [Parameter(Mandatory = $false, ParameterSetName = "Disabled")]
+      [switch] $Disable,
+
+      [ValidateRange(0, [int]::MaxValue)]
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [int] $JobTimeoutInMinutes,
+
+      [ValidateRange(0, 100)]
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [int] $MaxConcurrentAgentsPercentage,
+
+      [ValidateRange(0, [int]::MaxValue)]
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [int] $NumberOfHistoryRecordsToKeep,
+
+      [ValidateRange(0, [int]::MaxValue)]
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [int] $WorkingDirectoryExpirationInDays,
+
+      [ValidateRange(0, 23)]
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [int] $StartHours,
+
+      [ValidateRange(0, 59)]
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [int] $StartMinutes,
+
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [vsteam_lib.TimeZoneValidateAttribute()]
+      [ArgumentCompleter([vsteam_lib.TimeZoneCompleter])]
+      [string] $TimeZoneId,
+
+      [Parameter(Mandatory = $true, ParameterSetName = "Enabled")]
+      [vsteam_lib.AgentPoolMaintenanceDays] $WeekDaysToBuild
+   )
+
+   process {
+
+      if ($force -or $pscmdlet.ShouldProcess($Id, "Set Pool Maintenance")) {
+
+         $isEnabled = $true
+         if ($Disable.IsPresent) {
+            $isEnabled = $false
+         }else {
+
+         }
+
+         $resp = _callAPI -Method Get -NoProject -Area distributedtask -Resource pools -Id "$Id/maintenancedefinitions" -Version $(_getApiVersion DistributedTask)
```

Comment:
@DarqueWarrior  hope you don't mind. That I made this cmdlet to handle adding and updating a maintenance schedule in one command. Technically you can add only one, but there are only two states:

* no existing schedule --> add one
* update existing schedule

So I made it as you can just set the schedule. Also, activated/deactivated the schedule to follow the same experience as in the GUI, which makes more sense to me.